--- gdal-3.0.1/gcore/rasterio.cpp~	2019-06-28 12:58:14.000000000 +0200
+++ gdal-3.0.1/gcore/rasterio.cpp	2019-10-30 10:08:18.083471282 +0100
@@ -2872,15 +2872,15 @@
 
 void GDALUnrolledCopy_GByte_2_1_SSSE3( GByte* CPL_RESTRICT pDest,
                                              const GByte* CPL_RESTRICT pSrc,
-                                             GPtrDiff_t nIters );
+                                             GInt64 nIters );
 
 void GDALUnrolledCopy_GByte_3_1_SSSE3( GByte* CPL_RESTRICT pDest,
                                              const GByte* CPL_RESTRICT pSrc,
-                                             GPtrDiff_t nIters );
+                                             GInt64 nIters );
 
 void GDALUnrolledCopy_GByte_4_1_SSSE3( GByte* CPL_RESTRICT pDest,
                                              const GByte* CPL_RESTRICT pSrc,
-                                             GPtrDiff_t nIters );
+                                             GInt64 nIters );
 #endif
 
 
