--- gdal-1.1.8/configure.in.orig	2003-05-16 00:05:10.000000000 +0200
+++ gdal-1.1.8/configure.in	2003-05-16 00:10:51.000000000 +0200
@@ -747,8 +747,13 @@
     fi
   fi
 
-  echo '#include <util/PlatformUtils.hpp>' > conftest.cpp
-  echo 'int main() {  XMLPlatformUtils::Initialize(); }' >> conftest.cpp
+  cat > conftest.cpp <<EOF
+#include <util/PlatformUtils.hpp>
+#ifdef XERCES_CPP_NAMESPACE_USE
+XERCES_CPP_NAMESPACE_USE
+#endif
+int main() {  XMLPlatformUtils::Initialize(); }
+EOF
   CMD="${CXX} ${XERCES_INCLUDE} ${CPPFLAGS} -o conftest conftest.cpp $LIBS"
 
   if test -z "`$CMD 2>&1`" ; then
