--- gdal-1.9.0/swig/ruby/RubyMakefile.mk~	2012-01-04 08:03:41.000000000 +0100
+++ gdal-1.9.0/swig/ruby/RubyMakefile.mk	2012-02-13 16:12:17.778605160 +0100
@@ -41,7 +41,8 @@
 RUBY_LIB := -l$(RUBY_SO_NAME)
 else
 RUBY_MODULES=$(RUBY_MODULES_LIN)
-LDFLAGS += -Xcompiler -shared -L$(RUBY_LIB_DIR)
+LDFLAGS += -fPIC -Xcompiler -shared -L$(RUBY_LIB_DIR)
+CFLAGS += -fPIC
 RUBY_LIB := -l$(RUBY_SO_NAME)
 endif
 
