--- gdal-1.7.3/Doxyfile.orig	2010-11-07 19:28:49.000000000 +0100
+++ gdal-1.7.3/Doxyfile	2010-11-26 08:22:10.654873851 +0100
@@ -211,7 +211,7 @@
 # defined locally in source files will be included in the documentation. 
 # If set to NO only classes defined in header files are included.
 
-EXTRACT_LOCAL_CLASSES  = YES
+EXTRACT_LOCAL_CLASSES  = NO
 
 # If the HIDE_UNDOC_MEMBERS tag is set to YES, Doxygen will hide all 
 # undocumented members of documented classes, files or namespaces. 
