--- gdal-2.4.0/frmts/pdf/pdfdataset.cpp.orig	2018-12-14 22:37:15.000000000 +0100
+++ gdal-2.4.0/frmts/pdf/pdfdataset.cpp	2018-12-27 05:35:39.755925135 +0100
@@ -3459,7 +3459,7 @@
 #endif
             if( ocg != nullptr && ocg->getName() != nullptr )
             {
-                const char* pszLayerName = (const char*)ocg->getName()->getCString();
+                const char* pszLayerName = (const char*)ocg->getName()->c_str();
                 AddLayer(pszLayerName);
                 oLayerOCGMapPoppler[pszLayerName] = ocg;
             }
@@ -4787,7 +4787,7 @@
     GooString* poMetadata = poCatalogPoppler->readMetadata();
     if (poMetadata)
     {
-        const char* pszContent = poMetadata->getCString();
+        const char* pszContent = poMetadata->c_str();
         if (pszContent != nullptr &&
             STARTS_WITH(pszContent, "<?xpacket begin="))
         {
--- gdal-2.4.0/frmts/pdf/pdfobject.cpp.orig	2018-12-14 22:37:15.000000000 +0100
+++ gdal-2.4.0/frmts/pdf/pdfobject.cpp	2018-12-27 16:31:17.472142636 +0100
@@ -1061,7 +1061,7 @@
 #else
         GooString* gooString = m_po->getString();
 #endif
-        return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
+        return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->c_str()),
                                                       static_cast<int>(gooString->getLength())));
     }
     else
@@ -1422,7 +1422,7 @@
         char* pszContent = (char*) VSIMalloc(m_nLength + 1);
         if (pszContent)
         {
-            memcpy(pszContent, gstr->getCString(), m_nLength);
+            memcpy(pszContent, gstr->c_str(), m_nLength);
             pszContent[m_nLength] = '\0';
         }
         delete gstr;
