--- gdal-1.9.2/swig/ruby/ogr_wrap.cpp.orig	2012-11-18 19:58:26.045113264 +0100
+++ gdal-1.9.2/swig/ruby/ogr_wrap.cpp	2012-11-18 20:20:52.904032656 +0100
@@ -4264,7 +4264,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4327,7 +4327,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5152,7 +5152,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -5601,7 +5601,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5677,7 +5677,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5753,7 +5753,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5824,7 +5824,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5887,7 +5887,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6088,7 +6088,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6259,7 +6259,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6330,7 +6330,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6409,7 +6409,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6493,7 +6493,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6597,7 +6597,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6660,7 +6660,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6723,7 +6723,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6786,7 +6786,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6969,7 +6969,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7247,7 +7247,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7316,7 +7316,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8928,7 +8928,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -10334,7 +10334,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -10439,7 +10439,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -13371,7 +13371,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -13477,7 +13477,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -13884,7 +13884,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -13960,7 +13960,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -16769,7 +16769,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -16845,7 +16845,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -17677,7 +17677,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
--- gdal-1.9.2/swig/ruby/osr_wrap.cpp~	2012-11-18 19:48:51.125133547 +0100
+++ gdal-1.9.2/swig/ruby/osr_wrap.cpp	2012-11-18 20:40:20.190658146 +0100
@@ -2587,7 +2587,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -2659,7 +2659,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -3219,7 +3219,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3330,7 +3330,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3390,7 +3390,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3481,7 +3481,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3541,7 +3541,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3599,7 +3599,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -3771,7 +3771,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -3868,7 +3868,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc4 == SWIG_NEWOBJ) delete[] buf4;
@@ -3904,7 +3904,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -3947,7 +3947,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -4005,7 +4005,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -4113,7 +4113,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -4322,7 +4322,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4391,7 +4391,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4460,7 +4460,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4529,7 +4529,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4598,7 +4598,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4683,7 +4683,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4744,7 +4744,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4805,7 +4805,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4874,7 +4874,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -4951,7 +4951,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5028,7 +5028,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5089,7 +5089,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5150,7 +5150,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5184,7 +5184,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5253,7 +5253,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5322,7 +5322,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5415,7 +5415,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5516,7 +5516,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5609,7 +5609,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5678,7 +5678,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5763,7 +5763,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5840,7 +5840,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5925,7 +5925,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -5994,7 +5994,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6071,7 +6071,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6132,7 +6132,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6201,7 +6201,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6278,7 +6278,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6347,7 +6347,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6416,7 +6416,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6493,7 +6493,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6554,7 +6554,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6615,7 +6615,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6692,7 +6692,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6761,7 +6761,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6838,7 +6838,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -6924,7 +6924,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -6995,7 +6995,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7072,7 +7072,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7133,7 +7133,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7181,7 +7181,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7231,7 +7231,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7280,7 +7280,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7378,7 +7378,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7418,7 +7418,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7474,7 +7474,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7595,7 +7595,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7655,7 +7655,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7707,7 +7707,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7775,7 +7775,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7853,7 +7853,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7897,7 +7897,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -7940,7 +7940,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -7990,7 +7990,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -8040,7 +8040,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8092,7 +8092,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8134,7 +8134,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8215,7 +8215,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -8308,7 +8308,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8351,7 +8351,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -8424,7 +8424,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -8473,7 +8473,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (alloc2 == SWIG_NEWOBJ) delete[] buf2;
@@ -8515,7 +8515,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8589,7 +8589,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8653,7 +8653,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8729,7 +8729,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -8852,7 +8852,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   if (SWIG_IsTmpObj(res2)) {
@@ -8944,7 +8944,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9010,7 +9010,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9116,7 +9116,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9150,7 +9150,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9184,7 +9184,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9218,7 +9218,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9252,7 +9252,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
@@ -9286,7 +9286,7 @@
   {
     /* %typemap(out) OGRErr */
     if (result != 0) {
-      rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+      rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
     }
   }
   {
