--- gdal-1.3.2/frmts/fit/fitdataset.cpp.orig	2005-07-05 19:10:52.000000000 +0200
+++ gdal-1.3.2/frmts/fit/fitdataset.cpp	2006-03-05 13:07:26.440480750 +0100
@@ -1016,7 +1016,7 @@
         CPLDebug("FIT", "Loading file with header version 01");
 
         // map old style header into new header structure
-	FIThead01* head01 = (FIThead01*)&head;
+	FIThead01* head01 = (FIThead01*)head;
         gst_swapb(head->dataOffset);
 	info->dataOffset = head01->dataOffset;
 
--- gdal-1.3.2/ogr/ogrsf_frmts/ili/ogrili2layer.cpp.orig	2005-11-21 15:56:31.000000000 +0100
+++ gdal-1.3.2/ogr/ogrsf_frmts/ili/ogrili2layer.cpp	2006-06-03 12:49:56.937947000 +0200
@@ -73,7 +73,7 @@
     poFeatureDefn->SetGeomType( eReqType );
 
     bWriter = bWriterIn;
-    listFeatureIt = 0;
+    listFeatureIt = listFeature.begin();
 }
 
 /************************************************************************/
@@ -132,7 +132,6 @@
 /************************************************************************/
 
 OGRFeature *OGRILI2Layer::GetNextFeature() {
-    if (listFeatureIt == 0) listFeatureIt = listFeature.begin();
     if (listFeatureIt != listFeature.end())
         return *(listFeatureIt++);
     return NULL;
