Index: /trunk/gdal/configure
===================================================================
--- /trunk/gdal/configure	(revision 26534)
+++ /trunk/gdal/configure	(revision 26582)
@@ -723,4 +723,5 @@
 II_SYSTEM
 HAVE_INGRES
+MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION
 MYSQL_LIB
 MYSQL_INC
@@ -23308,4 +23309,32 @@
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
+
+      # Check if mysql headers declare load_defaults
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking load_defaults() in MySQL" >&5
+$as_echo_n "checking load_defaults() in MySQL... " >&6; }
+      rm -f testmysql.*
+      echo '#include "my_global.h"' > testmysql.cpp
+      echo '#include "my_sys.h"' >> testmysql.cpp
+      echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
+      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes, found in my_sys.h" >&5
+$as_echo "yes, found in my_sys.h" >&6; }
+      else
+        echo 'extern "C" void load_defaults(const char *conf_file, const char **groups, int *argc, char ***argv);' > testmysql.cpp
+        echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
+        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+            { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes, found in library but not in header" >&5
+$as_echo "yes, found in library but not in header" >&6; }
+            MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION=yes
+        else
+            HAVE_MYSQL=no
+            MYSQL_LIB=
+            MYSQL_INC=
+            as_fn_error $? "Cannot find load_defaults()" "$LINENO" 5
+        fi
+      fi
+      rm -f testmysql.*
+      rm -f testmysql
+
 	;;
   esac
@@ -23317,4 +23346,6 @@
 
 MYSQL_LIB=$MYSQL_LIB
+
+MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION=$MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION
 
 
Index: /trunk/gdal/configure.in
===================================================================
--- /trunk/gdal/configure.in	(revision 26534)
+++ /trunk/gdal/configure.in	(revision 26582)
@@ -2462,4 +2462,29 @@
       MYSQL_INC="`$MYSQL_CONFIG --include`"
       AC_MSG_RESULT([yes])
+
+      # Check if mysql headers declare load_defaults
+      AC_MSG_CHECKING([load_defaults() in MySQL])
+      rm -f testmysql.*
+      echo '#include "my_global.h"' > testmysql.cpp
+      echo '#include "my_sys.h"' >> testmysql.cpp
+      echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
+      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+        AC_MSG_RESULT([yes, found in my_sys.h])
+      else
+        echo 'extern "C" void load_defaults(const char *conf_file, const char **groups, int *argc, char ***argv);' > testmysql.cpp
+        echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
+        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+            AC_MSG_RESULT([yes, found in library but not in header])
+            MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION=yes
+        else
+            HAVE_MYSQL=no
+            MYSQL_LIB=
+            MYSQL_INC=
+            AC_MSG_ERROR([Cannot find load_defaults()])
+        fi
+      fi
+      rm -f testmysql.*
+      rm -f testmysql
+
 	;;
   esac
@@ -2469,4 +2494,5 @@
 AC_SUBST(MYSQL_INC,$MYSQL_INC)
 AC_SUBST(MYSQL_LIB,$MYSQL_LIB)
+AC_SUBST(MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION,$MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION)
 
 dnl ---------------------------------------------------------------------------
Index: /trunk/gdal/ogr/ogrsf_frmts/mysql/GNUmakefile
===================================================================
--- /trunk/gdal/ogr/ogrsf_frmts/mysql/GNUmakefile	(revision 15888)
+++ /trunk/gdal/ogr/ogrsf_frmts/mysql/GNUmakefile	(revision 26582)
@@ -8,4 +8,9 @@
 CPPFLAGS	:=	-I.. -I../.. $(GDAL_INCLUDE) $(MYSQL_INC) $(CPPFLAGS)
 
+ifeq ($(MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION),yes)
+CPPFLAGS +=   -DMYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION
+endif
+
+
 default:	$(O_OBJ:.o=.$(OBJ_EXT))
 
Index: /trunk/gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
===================================================================
--- /trunk/gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	(revision 26506)
+++ /trunk/gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	(revision 26582)
@@ -37,4 +37,14 @@
 #include "cpl_string.h"
 
+/* Recent versions of mysql no longer declare load_defaults() in my_sys.h */
+/* but they still have it in the lib. Very fragile... */
+#ifdef MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION
+extern "C" {
+int load_defaults(const char *conf_file, const char **groups,
+                  int *argc, char ***argv);
+void free_defaults(char **argv);
+}
+#endif
+
 CPL_CVSID("$Id$");
 /************************************************************************/
Index: /trunk/gdal/GDALmake.opt.in
===================================================================
--- /trunk/gdal/GDALmake.opt.in	(revision 26296)
+++ /trunk/gdal/GDALmake.opt.in	(revision 26582)
@@ -127,4 +127,5 @@
 MYSQL_LIB  =	@MYSQL_LIB@
 MYSQL_INC  =	@MYSQL_INC@
+MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION  =    @MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION@
 LIBS	   +=	$(MYSQL_LIB)
 
Index: /trunk/gdal/configure
===================================================================
--- /trunk/gdal/configure	(revision 26582)
+++ /trunk/gdal/configure	(revision 26583)
@@ -23317,5 +23317,5 @@
       echo '#include "my_sys.h"' >> testmysql.cpp
       echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
-      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} -o testmysql testmysql.cpp ${MYSQL_LIB} 2>&1`" ; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes, found in my_sys.h" >&5
 $as_echo "yes, found in my_sys.h" >&6; }
@@ -23323,5 +23323,5 @@
         echo 'extern "C" void load_defaults(const char *conf_file, const char **groups, int *argc, char ***argv);' > testmysql.cpp
         echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
-        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} -o testmysql testmysql.cpp ${MYSQL_LIB} 2>&1`" ; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes, found in library but not in header" >&5
 $as_echo "yes, found in library but not in header" >&6; }
Index: /trunk/gdal/configure.in
===================================================================
--- /trunk/gdal/configure.in	(revision 26582)
+++ /trunk/gdal/configure.in	(revision 26583)
@@ -2469,10 +2469,10 @@
       echo '#include "my_sys.h"' >> testmysql.cpp
       echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
-      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+      if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} -o testmysql testmysql.cpp ${MYSQL_LIB} 2>&1`" ; then
         AC_MSG_RESULT([yes, found in my_sys.h])
       else
         echo 'extern "C" void load_defaults(const char *conf_file, const char **groups, int *argc, char ***argv);' > testmysql.cpp
         echo 'int main(int argc, char** argv) { load_defaults(0, 0, 0, 0); return 0; } ' >> testmysql.cpp
-        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} ${MYSQL_LIB} testmysql.cpp -o testmysql 2>&1`" ; then
+        if test -z "`${CXX} ${CXXFLAGS} ${MYSQL_INC} -o testmysql testmysql.cpp ${MYSQL_LIB} 2>&1`" ; then
             AC_MSG_RESULT([yes, found in library but not in header])
             MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION=yes
