--- gdal-1.1.7/aclocal.m4.orig	Thu Jan 17 02:40:48 2002
+++ gdal-1.1.7/aclocal.m4	Mon Jun 17 22:24:56 2002
@@ -395,10 +395,12 @@
     py_makefile="`$PYTHON -c '
 import sys
 print \"%s/lib/python%s/config/Makefile\"%(sys.exec_prefix, sys.version[:3])'`"
+    changequote([, ])
     if test ! -f "$py_makefile"; then
       AC_MSG_ERROR([*** Couldn't find the python config makefile.  Maybe you are
 *** missing the development portion of the python installation])
     fi
+    changequote(,)dnl
     eval `sed -n \
 -e "s/^CC=[ 	]*\(.*\)/am_cv_python_CC='\1'/p" \
 -e "s/^OPT=[ 	]*\(.*\)/am_cv_python_OPT='\1'/p" \
