--- gdal-1.9.2/swig/include/ruby/typemaps_ruby.i.orig	2012-10-09 02:58:28.000000000 +0200
+++ gdal-1.9.2/swig/include/ruby/typemaps_ruby.i	2013-01-06 15:12:18.221272187 +0100
@@ -127,7 +127,7 @@
 {
   /* %typemap(out) OGRErr */
   if ($1 != 0) {
-    rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+    rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
   }
 }
 
--- gdal-1.11.0/ogr/ogrsf_frmts/sxf/ogrsxfdatasource.cpp.orig	2014-04-16 22:04:02.000000000 +0200
+++ gdal-1.11.0/ogr/ogrsf_frmts/sxf/ogrsxfdatasource.cpp	2014-05-10 17:33:52.404601057 +0200
@@ -459,7 +459,7 @@
     if (nEPSG == 0)
     {
         CPLError( CE_Warning, CPLE_NotSupported,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld) not supported", iVCS) );
+                  "SXF. Vertical coordinate system (SXF index %ld) not supported", iVCS );
         return;
     }
 
@@ -468,14 +468,14 @@
     if (eImportFromEPSGErr != OGRERR_NONE)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld, EPSG %d) import from EPSG error", iVCS, nEPSG) );
+                  "SXF. Vertical coordinate system (SXF index %ld, EPSG %d) import from EPSG error", iVCS, nEPSG );
         return;
     }
 
     if (sr->IsVertical() != 1)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Coordinate system (SXF index %ld, EPSG %d) is not Vertical", iVCS, nEPSG) );
+                  "SXF. Coordinate system (SXF index %ld, EPSG %d) is not Vertical", iVCS, nEPSG );
         return;
     }
 
@@ -484,7 +484,7 @@
     if (eSetVertCSErr != OGRERR_NONE)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld, EPSG %d) set error", iVCS, nEPSG) );
+                  "SXF. Vertical coordinate system (SXF index %ld, EPSG %d) set error", iVCS, nEPSG );
         return;
     }
 }
