--- gdal-1.11.0/configure.in.orig	2014-05-12 21:49:57.910679098 +0200
+++ gdal-1.11.0/configure.in	2014-05-18 11:22:22.873992206 +0200
@@ -1480,23 +1480,11 @@
 
 elif test "$with_sosi" = "yes" -o "$with_sosi" = "" ; then
 
-    AC_MSG_CHECKING([for libfyba.a, libfygm.a and libfyut.a in /usr/local/lib])
-    if test -r /usr/local/lib/libfyba.a -a -r /usr/local/lib/libfygm.a -a -r /usr/local/lib/libfyut.a ; then
-        AC_MSG_RESULT([found])
+    AC_CHECK_LIB([fyba], [main], [
         SOSI_ENABLED=yes
-        SOSI_LIB="/usr/local/lib/libfyba.a /usr/local/lib/libfygm.a /usr/local/lib/libfyut.a"
-    else
-        AC_MSG_ERROR([not found.])
-    fi
-
-    AC_MSG_CHECKING([for fyba.h, fygm.h and fyut.h in /usr/local/include/fyba])
-    if test -r /usr/local/include/fyba/fyba.h -a -r /usr/local/include/fyba/fygm.h -a -r /usr/local/include/fyba/fyut.h ; then
-        AC_MSG_RESULT([found])
-        SOSI_ENABLED=yes
-        SOSI_INC="-I/usr/local/include/fyba"
-    else
-        AC_MSG_ERROR([not found.])
-    fi
+        SOSI_INC="-I/usr/include/fyba"
+        SOSI_LIB="-lfyba -lfygm -lfyut"
+    ], [AC_MSG_ERROR([not found.])])
 
 else
 
