--- gdal-1.9.2/swig/include/ruby/typemaps_ruby.i.orig	2012-10-09 02:58:28.000000000 +0200
+++ gdal-1.9.2/swig/include/ruby/typemaps_ruby.i	2013-01-06 15:12:18.221272187 +0100
@@ -127,7 +127,7 @@
 {
   /* %typemap(out) OGRErr */
   if ($1 != 0) {
-    rb_raise(rb_eRuntimeError, OGRErrMessages(result));
+    rb_raise(rb_eRuntimeError, "%s", OGRErrMessages(result));
   }
 }
 
--- gdal-1.11.2/ogr/ogrsf_frmts/sxf/ogrsxfdatasource.cpp.orig	2015-02-10 13:11:16.000000000 +0100
+++ gdal-1.11.2/ogr/ogrsf_frmts/sxf/ogrsxfdatasource.cpp	2015-02-28 12:20:28.595919291 +0100
@@ -458,7 +458,7 @@
     if (nEPSG == 0)
     {
         CPLError( CE_Warning, CPLE_NotSupported,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld) not supported", iVCS) );
+                  "SXF. Vertical coordinate system (SXF index %ld) not supported", iVCS );
         return;
     }
 
@@ -467,14 +467,14 @@
     if (eImportFromEPSGErr != OGRERR_NONE)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld, EPSG %ld) import from EPSG error", iVCS, nEPSG) );
+                  "SXF. Vertical coordinate system (SXF index %ld, EPSG %ld) import from EPSG error", iVCS, nEPSG );
         return;
     }
 
     if (sr->IsVertical() != 1)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Coordinate system (SXF index %ld, EPSG %ld) is not Vertical", iVCS, nEPSG) );
+                  "SXF. Coordinate system (SXF index %ld, EPSG %ld) is not Vertical", iVCS, nEPSG );
         return;
     }
 
@@ -483,7 +483,7 @@
     if (eSetVertCSErr != OGRERR_NONE)
     {
         CPLError( CE_Warning, CPLE_None,
-                  CPLString().Printf("SXF. Vertical coordinate system (SXF index %ld, EPSG %ld) set error", iVCS, nEPSG) );
+                  "SXF. Vertical coordinate system (SXF index %ld, EPSG %ld) set error", iVCS, nEPSG );
         return;
     }
 }
--- gdal-1.11.2/ogr/ogrsf_frmts/oci/ogrociwritablelayer.cpp.orig	2015-02-10 13:11:16.000000000 +0100
+++ gdal-1.11.2/ogr/ogrsf_frmts/oci/ogrociwritablelayer.cpp	2015-02-28 13:46:28.845702734 +0100
@@ -321,7 +321,7 @@
                           + strlen(oField.GetNameRef())
                           + strlen(szFieldType) );
 
-    snprintf( szFieldName, sizeof( szFieldName ), oField.GetNameRef());
+    snprintf( szFieldName, sizeof( szFieldName ), "%s", oField.GetNameRef());
     szFieldName[sizeof( szFieldName )-1] = '\0';
     if ( strlen(oField.GetNameRef()) > sizeof ( szFieldName ) )
     {
