--- gdal-1.11.0/configure.ac.orig	2014-06-14 20:12:03.704706951 +0200
+++ gdal-1.11.0/configure.ac	2014-06-14 20:53:44.987988150 +0200
@@ -4182,7 +4182,7 @@
   AC_CHECK_LIB(raslib,main,RASDAMAN_ENABLED=yes,,,)
 
   if test -n "$RASDAMAN_ENABLED" ; then
-      RASDAMAN_LIB="-lrasodmg -lclientcomm -lcompression -lnetwork -lraslib"
+      RASDAMAN_LIB="-lrasodmg -lclientcomm -lnetwork -lraslib"
       OPT_GDAL_FORMATS="rasdaman $OPT_GDAL_FORMATS"
       AC_MSG_RESULT([        using rasdaman from system libs.])
   else
@@ -4195,7 +4195,7 @@
 
   if test -f $RASDAMAN_DIR/include/rasdaman.hh -a -d$RASDAMAN_DIR/lib ; then
       RASDAMAN_INC=-I$RASDAMAN_DIR/include
-      RASDAMAN_LIB="-L$RASDAMAN_DIR/lib -lrasodmg -lclientcomm -lcompression -lnetwork -lraslib"
+      RASDAMAN_LIB="-L$RASDAMAN_DIR/lib -lrasodmg -lclientcomm -lnetwork -lraslib"
   else
       AC_MSG_ERROR("Could not find rasdaman.hh or libraslib.a in $RASDAMAN_DIR.")
   fi
