diff -urNp -x '*.orig' gdal-3.0.4.org/configure.ac gdal-3.0.4/configure.ac
--- gdal-3.0.4.org/configure.ac	2021-04-02 16:52:15.663388370 +0200
+++ gdal-3.0.4/configure.ac	2021-04-02 16:52:16.026721386 +0200
@@ -5219,12 +5219,16 @@ dnl ------------------------------------
 
 if test "$prefix" = "NONE" ; then
   GDAL_PREFIX=/usr/local
+  GDAL_LIBDIR=${GDAL_PREFIX}/lib
 else
   GDAL_PREFIX=${prefix}
+  eval eval GDAL_LIBDIR="${libdir}"
 fi
 
 export GDAL_PREFIX
+export GDAL_LIBDIR
 AC_DEFINE_UNQUOTED(GDAL_PREFIX,["]$GDAL_PREFIX["], [--prefix directory for GDAL install])
+AC_DEFINE_UNQUOTED(GDAL_LIBDIR,["]$GDAL_LIBDIR["], [--prefix directory for GDAL install])
 
 dnl ---------------------------------------------------------------------------
 dnl Perl bindings.
diff -urNp -x '*.orig' gdal-3.0.4.org/gcore/gdaldrivermanager.cpp gdal-3.0.4/gcore/gdaldrivermanager.cpp
--- gdal-3.0.4.org/gcore/gdaldrivermanager.cpp	2020-01-28 11:18:44.000000000 +0100
+++ gdal-3.0.4/gcore/gdaldrivermanager.cpp	2021-04-02 16:52:16.026721386 +0200
@@ -745,12 +745,12 @@ void GDALDriverManager::AutoLoadDrivers(
     }
     else
     {
-#ifdef GDAL_PREFIX
+#if 1
         papszSearchPath = CSLAddString( papszSearchPath,
     #ifdef MACOSX_FRAMEWORK
                                         GDAL_PREFIX "/PlugIns");
     #else
-                                        GDAL_PREFIX "/lib/gdalplugins" );
+                                        GDAL_LIBDIR "/gdalplugins" );
     #endif
 #else
         char szExecPath[1024];
