--- gdal-1.3.2/configure.in.orig	2006-05-01 18:22:48.000000000 +0200
+++ gdal-1.3.2/configure.in	2006-05-29 01:24:48.086491750 +0200
@@ -275,7 +275,7 @@
   PG_EXTRA_DEC=""
   AC_CHECK_LIB(pq,PQputCopyData,,PG_EXTRA_DEC=-DPG_PRE74,)
 
-  PG_INC="-I`$PG_CONFIG --includedir` $PG_EXTRA_DEC"
+  PG_INC="-I`$PG_CONFIG --includedir` -I`$PG_CONFIG --includedir-server` $PG_EXTRA_DEC"
 fi
 
 AC_SUBST(HAVE_PG,$HAVE_PG)
