--- gdal-1.3.2/configure.in.orig	2006-12-11 23:03:29.053886000 +0100
+++ gdal-1.3.2/configure.in	2006-12-11 23:08:09.409862968 +0100
@@ -254,7 +254,7 @@
   PG_LIB=
   PG_INC=
 
-  AC_MSG_RESULT([no])LIBZ
+  AC_MSG_RESULT([no])
 
 else
   if test -d $PG_CONFIG ; then
@@ -1517,7 +1517,7 @@
     else
         LIBS="$LIBS -L$with_sqlite/lib -lsqlite3"
         SQLITE_INC="-I$with_sqlite/include"
-        AC_MSG_RESULT(setting DODS root directory to $with_sqlite)_
+        AC_MSG_RESULT(setting DODS root directory to $with_sqlite)
     fi
     HAVE_SQLITE=yes
 fi
--- gdal-1.3.2/m4/acinclude.m4.orig	2006-04-24 19:19:28.000000000 +0200
+++ gdal-1.3.2/m4/acinclude.m4	2006-12-11 23:25:37.569594186 +0100
@@ -452,10 +452,9 @@
   fi
 
   ARCH=`uname -i 2>/dev/null`
-  PYLIB=lib
-  if test "$ARCH" = "x86_64" ; then
-    PYLIB=lib64
-  fi
+  PYLIB="`$PYTHON -c '
+import sys
+print sys.lib'`"
 
   if test "$PYTHON" != no ; then
     AC_MSG_CHECKING([where python Makefiles are])
@@ -529,14 +528,14 @@
   AC_MSG_CHECKING([for python headers])
   AC_CACHE_VAL(am_cv_python_includes,
     [changequote(,)dnl
-    am_cv_python_includes="`$PYTHON -c '
+    am_cv_python_includes="$($PYTHON -c '
 import sys
-includepy = \"%s/include/python%s\" % (sys.prefix, sys.version[:3])
-if sys.version[0] > \"1\" or sys.version[2] > \"4\":
-  libpl = \"%s/include/python%s\" % (sys.exec_prefix, sys.version[:3])
+includepy = "%s/include/python%s" % (sys.prefix, sys.version[:3])
+if sys.version[0] > "1" or sys.version[2] > "4":
+  libpl = "%s/include/python%s" % (sys.exec_prefix, sys.version[:3])
 else:
-  libpl = \"%s/'$PYLIB'/python%s/config\" % (sys.exec_prefix, sys.version[:3])
-print \"-I%s -I%s\" % (includepy, libpl)'`"
+  libpl = "%s/'$PYLIB'/python%s/config" % (sys.exec_prefix, sys.version[:3])
+print "-I%s -I%s" % (includepy, libpl)')"
     changequote([, ])])
   PYTHON_INCLUDES="$am_cv_python_includes"
   AC_MSG_RESULT(found)
@@ -546,13 +545,13 @@
   AC_CACHE_VAL(am_cv_python_makefile,
     [changequote(,)dnl
     if test ! -z "`uname -a | grep CYGWIN`" ; then 
-      PYTHON_LIBS="`$PYTHON -c '
+      PYTHON_LIBS="$($PYTHON -c '
 import sys
-print \"-L%s/'$PYLIB'/python%s/config -lpython%s.dll\" % (sys.prefix, sys.version[:3], sys.version[:3])'`"
+print "-L%s/'$PYLIB'/python%s/config -lpython%s.dll" % (sys.prefix, sys.version[:3], sys.version[:3])')"
     fi 
-    py_makefile="`$PYTHON -c '
+    py_makefile="$($PYTHON -c '
 import sys
-print \"%s/'$PYLIB'/python%s/config/Makefile\"%(sys.exec_prefix, sys.version[:3])'`"
+print "%s/'$PYLIB'/python%s/config/Makefile"%(sys.exec_prefix, sys.version[:3])')"
     if test ! -f "$py_makefile"; then
       echo Could not find the python config makefile.  Maybe you are;
       echo missing the development portion of the python installation;
