--- gdal-1.1.8/GNUmakefile.orig	Sun Jan 26 12:36:20 2003
+++ gdal-1.1.8/GNUmakefile	Sun Jan 26 13:39:08 2003
@@ -10,7 +10,7 @@
 force-lib:	
 	$(AR) r $(GDAL_LIB) $(GDAL_OBJ)
 	$(RANLIB) $(GDAL_LIB)
-	$(LD_SHARED) $(GDAL_OBJ) $(GDAL_LIBS) $(LIBS) -o $(GDAL_SLIB)
+	$(LD_SHARED) -Wl,-soname=libgdal.$(GDAL_VER).$(SO_EXT) $(GDAL_OBJ) $(GDAL_LIBS) $(LIBS) -o $(GDAL_SLIB)
 
 #	If you really want proper SO files that will work in /usr/lib
 # 	Try replacing the above command with something like this:
@@ -23,7 +23,7 @@
 	$(RANLIB) $(GDAL_LIB)
 
 $(GDAL_SLIB):	$(GDAL_LIB) $(GDALA_DEP)
-	$(LD_SHARED) $(GDAL_OBJ) $(GDAL_LIBS) $(LIBS) -o $(GDAL_SLIB)
+	$(LD_SHARED) -Wl,-soname=libgdal.$(GDAL_VER).$(SO_EXT) $(GDAL_OBJ) $(GDAL_LIBS) $(LIBS) -o $(GDAL_SLIB)
 
 check-lib:
 	$(MAKE) $(GDAL_LIB)
@@ -143,6 +143,7 @@
 	$(INSTALL) $(GDAL_LIB) $(DESTDIR)$(INST_LIB)
 ifeq ($(HAVE_LD_SHARED),yes)
 	$(INSTALL) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)
+	ln -sf $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/libgdal.so
 endif
 	for f in data/*.* ; do $(INSTALL) -m 0644 $$f $(DESTDIR)$(INST_DATA) ; done
 
