--- gdal-2.0.2/m4/lib-prefix.m4~	2016-01-26 16:21:45.000000000 +0100
+++ gdal-2.0.2/m4/lib-prefix.m4	2017-09-04 23:14:12.632192795 +0200
@@ -172,9 +172,11 @@
     for searchdir in $searchpath; do
       if test -d "$searchdir"; then
         case "$searchdir" in
+          */libx32/ | */libx32 ) acl_libdirstem=libx32 ;;
           */lib64/ | */lib64 ) acl_libdirstem=lib64 ;;
           *) searchdir=`cd "$searchdir" && pwd`
              case "$searchdir" in
+               */libx324 ) acl_libdirstem=libx32 ;;
                */lib64 ) acl_libdirstem=lib64 ;;
              esac ;;
         esac
